src/main/java/com/minecolonies/core/entity/ai/workers/guard/training/EntityAIArcherTraining.java

```diff
@@ -111,7 +114,8 @@ private IAIState selectTarget()
             return DECIDE;
         }
         final BlockPos targetPos = archeryBuilding.getRandomShootingTarget(worker.getRandom());
-        if (targetPos == null)
+        if (targetPos == null || !WorldUtil.isBlockLoaded(world, targetPos) ||
+              !world.clip(new ClipContext(new Vec3(worker.getX(), worker.getEyeY(), worker.getZ()), new Vec3(targetPos.getX() + HALF_BLOCK, targetPos.getY() + HALF_BLOCK, targetPos.getZ() + HALF_BLOCK), ClipContext.Block.COLLIDER, net.minecraft.world.level.ClipContext.Fluid.NONE, worker)).getBlockPos().equals(targetPos))
```

Comment:
make sure to test it with different target blocks, it may be depending on the target shape that the check fails

Reply:
We only use full block target blocks here

Reply:
oh ok thought we had added tags

Reply:
for the shooting position only